libkbfs/dirty_bcache.go

```diff
@@ -67,44 +115,46 @@ type DirtyBlockCacheStandard struct {
 	// accepted. Used only for testing.
 	blockedChanForTesting chan<- int64
 
-	// The minimum (and initial) size of the sync buffer.
-	minSyncBufferSize int64
-	// The maximum size of the sync buffer.  Also used as the
+	// The minimum (and initial) capacity of the sync buffer.
+	minSyncBufCap int64
+	// The maximum capacity of the sync buffer.  Also used as the
 	// denominator when calculating backpressure, such that the closer
 	// we are to reaching the maximum size (over and above the current
 	// sync buffer), the more write requests will be delayed.
-	maxSyncBufferSize int64
+	maxSyncBufCap int64
 
 	shutdownLock sync.RWMutex
 	isShutdown   bool
 
-	lock               sync.RWMutex
-	cache              map[dirtyBlockID]Block
-	unsyncedDirtyBytes int64
-	syncingDirtyBytes  int64 // just for bookkeeping, not actually used
-	totalDirtyBytes    int64
-	syncBufferSize     int64
+	lock            sync.RWMutex
+	cache           map[dirtyBlockID]Block
+	syncBufBytes    int64
+	waitBufBytes    int64
+	syncBufferCap   int64
+	ignoreSyncBytes int64 // these bytes have "timed out"
+	syncStarted     time.Time
 }
 
 // NewDirtyBlockCacheStandard constructs a new BlockCacheStandard
 // instance.  makeLog is a function that will be called later to make
 // a log (we don't take an actual log, to allow the DirtyBlockCache to
 // be created before logging is initialized).  The min and max buffer
-// sizes define the possible range of how many bytes we'll try to sync
-// in any one sync.
+// capacities define the possible range of how many bytes we'll try to
+// sync in any one sync, and the start size defines the initial buffer
+// size.
 func NewDirtyBlockCacheStandard(clock Clock,
-	makeLog func(string) logger.Logger, minSyncBufferSize int64,
-	maxSyncBufferSize int64) *DirtyBlockCacheStandard {
+	makeLog func(string) logger.Logger, minSyncBufCap int64,
```

Comment:
might be better to make arg order match the order of declaration above and below